### ISSUE TITLE
Refactor network calls to use asynchronous API

### ISSUE THREAD

**Issue by amazuerar** [ultrasonic/ultrasonic#555]
## Problem description

Dear Developer!

My name is Alejandro Mazuera-Rozo, I am a PhD Student at Universidad de los Andes, and at Università della Svizzera italiana. I am part of a research on the usage of network libraries within Android apps. As result of this we identified some code locations that might have network related problems.

In this case, we address the code locations that are related to a potential issue because of implementing synchronous operations. As you might know, using the `execute()` method on a call object will perform a synchronous request which is not recommended and its usage is discouraged.

1. In **synchronous operations**, tasks are performed one at a time and only when one is completed, the following is unblocked.

2. In **asynchronous operations**, you can move to another task before the previous one finishes, allowing you to deal with multiple requests simultaneously.
           
Moreover, by avoiding this synchronous approach, it is less likely to perform network operations on the main thread. Android requires you to perform network operations on a thread other than the main UI thread; a `NetworkOnMainThreadException` is thrown otherwise.
       
In order to address this issue we recommend you to visit:

1. https://developer.android.com/training/basics/network-ops/connecting

Potential Code Location with synchronous operations

https://github.com/ultrasonic/ultrasonic/blob/6e6cf3a86a2efd3669cbb8c1e546545d256ded12/ultrasonic/src/main/kotlin/org/moire/ultrasonic/service/RESTMusicService.kt#L90

https://github.com/ultrasonic/ultrasonic/blob/6e6cf3a86a2efd3669cbb8c1e546545d256ded12/ultrasonic/src/main/kotlin/org/moire/ultrasonic/service/RESTMusicService.kt#L187

https://github.com/ultrasonic/ultrasonic/blob/6e6cf3a86a2efd3669cbb8c1e546545d256ded12/ultrasonic/src/main/kotlin/org/moire/ultrasonic/service/RESTMusicService.kt#L214

https://github.com/ultrasonic/ultrasonic/blob/6e6cf3a86a2efd3669cbb8c1e546545d256ded12/ultrasonic/src/main/kotlin/org/moire/ultrasonic/service/RESTMusicService.kt#L227

https://github.com/ultrasonic/ultrasonic/blob/6e6cf3a86a2efd3669cbb8c1e546545d256ded12/ultrasonic/src/main/kotlin/org/moire/ultrasonic/service/RESTMusicService.kt#L344

https://github.com/ultrasonic/ultrasonic/blob/6e6cf3a86a2efd3669cbb8c1e546545d256ded12/ultrasonic/src/main/kotlin/org/moire/ultrasonic/service/RESTMusicService.kt#L545

https://github.com/ultrasonic/ultrasonic/blob/6e6cf3a86a2efd3669cbb8c1e546545d256ded12/ultrasonic/src/main/kotlin/org/moire/ultrasonic/service/RESTMusicService.kt#L565

https://github.com/ultrasonic/ultrasonic/blob/6e6cf3a86a2efd3669cbb8c1e546545d256ded12/ultrasonic/src/main/kotlin/org/moire/ultrasonic/service/RESTMusicService.kt#L577

https://github.com/ultrasonic/ultrasonic/blob/6e6cf3a86a2efd3669cbb8c1e546545d256ded12/ultrasonic/src/main/kotlin/org/moire/ultrasonic/service/RESTMusicService.kt#L600

https://github.com/ultrasonic/ultrasonic/blob/6e6cf3a86a2efd3669cbb8c1e546545d256ded12/ultrasonic/src/main/kotlin/org/moire/ultrasonic/imageloader/AvatarRequestHandler.kt#L27

https://github.com/ultrasonic/ultrasonic/blob/6e6cf3a86a2efd3669cbb8c1e546545d256ded12/ultrasonic/src/main/kotlin/org/moire/ultrasonic/imageloader/CoverArtRequestHandler.kt#L43

### Steps to reproduce

N/A

## System information

N/A

## Additional notes

The issues were detected by running a static analysis tool, therefore, it is based on source code.

**Comment by nitehu** [comment]
Thank you for your work.
Ultrasonic has lots of legacy code so it is not trivial to refactor it to asynchronous operation yet.
However the issue is valid, maybe we'll find the time to work on it in the future.

**Comment by tzugen** [comment]
The functions in Restmusicservice are not async, but they are only called from inside coroutine so they are in fact not blocking
